js/mexc.js

```diff
@@ -2163,6 +2164,84 @@ module.exports = class mexc extends Exchange {
         }
     }
 
+    async fetchFundingHistory (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchFundingHistory() requires a symbol argument');
+        }
+        let pageNum = undefined;
+        let pageSize = undefined;
+        if (params['page_num'] === undefined) {
+            pageNum = 1;
+        } else {
+            pageNum = params['page_num'];
+        }
+        if (params['page_size'] === undefined) {
+            pageSize = 20;
+        } else {
+            pageSize = params['page_size'];
+        }
+        const request = {
```

Comment:
We should not do lines 2172-2183, cause the request is extended with params-overrides in `this.extend` right before sending it out to the exchange, so whatever overrides the user has specified in `params` – those get into the request and override the defaults. So, we should just fill the `request` with the defaults. Also, we should not send a parameter, if it is optional on the exchange side.

Reply:
Let me know if these changes are okay or not, it works with page_num and page_size commented out but the exchange api docs say that these fields are mandatory. I also removed the ArgumentsRequired for a symbol argument.

Reply:
Most of the changes are ok, but if the user has supplied a `symbol` argument and the exchange supports it – then we must send the market['id'] to the exchange in the request. We should not send the market['id'] if the user has not supplied a symbol argument.

Reply:
Sorry I missed that, I think it's good now

Reply:
Thanks!